lib/dynflow/clock.rb

```diff
@@ -84,18 +84,17 @@ def first_timer
     end
 
     def wakeup
-      @timer.stop if @timer
+      if @timer
+        @timer.cancel if @timer
+        tick unless terminating?
+      end
     end
 
     def sleep_to(timer)
       return unless timer
       sec = [timer.when - Time.now, 0.0].max
-      @timer = Concurrent::Channel.timer(sec)
-      Concurrent::Channel.go do
-        ~@timer
-        return if terminating?
-        reference.tell(:tick)
-      end
+      @timer = Concurrent::ScheduledTask.execute(sec) { reference.tell(:tick) unless terminating? }
```

Comment:
[Docs](https://github.com/ruby-concurrency/concurrent-ruby/blob/d11b29c37b81320ca7126cc9cd85f4f3d17a78a3/lib/concurrent/scheduled_task.rb#L22) are saying sec == 0 is going to fail. I haven't tested it though.

Reply:
Docs aren't completely right
```
> Concurrent::ScheduledTask.execute(-1) { puts "boom" }; nil
ArgumentError: seconds must be greater than zero
from /home/aruzicka/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/concurrent-ruby-1.1.5/lib/concurrent/scheduled_task.rb:167:in `initialize'

> Concurrent::ScheduledTask.execute(0) { puts "boom" }; nil
=> nil
boom
```

will fix anyway